students/Kwonminseok/users/views.py

```diff
@@ -1,3 +1,39 @@
-from django.shortcuts import render
+import json
+import re
 
-# Create your views here.
+from django.views           import View
+from django.http            import JsonResponse, HttpResponse
+from django.core.exceptions import ValidationError
+
+from .models import User
+
+class SignUpView(View):
+    def post(self, request):       
+        try:
+            data           = json.loads(request.body)
+            name           = data["name"]
+            email          = data["email"]
+            password       = data["password"]
+            phone_number   = data["phone_number"]
+            age            = data["age"]
+ 
+            if not (password and email):
+                return JsonResponse({'message' : 'NO_VALUE'}, status=400)
+            if not re.match('^[a-zA-Z0-9+-\_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$', email):
+                return JsonResponse({'message' : 'EMAIL_VALIDATION_ERROR'}, status=400)
+            if not re.match("^(?=.*[A-Za-z])(?=.*\d)(?=.*[$@$!%*#?&])[A-Za-z\d$@$!%*#?&]{8,}$", password):
+                return JsonResponse({'message' : 'PASSWORD_VALIDATION_ERROR'}, status=400)
+            if User.objects.filter(email=email).exists():
+                return JsonResponse({'message' : 'DUPLICATED EMAIL'}, status=400)
```

Comment:
if문 사이에는 한 줄씩 띄워주세요~

```suggestion
            if not (password and email):
                return JsonResponse({'message' : 'NO_VALUE'}, status=400)
                
            if not re.match('^[a-zA-Z0-9+-\_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$', email):
                return JsonResponse({'message' : 'EMAIL_VALIDATION_ERROR'}, status=400)
                
            if not re.match("^(?=.*[A-Za-z])(?=.*\d)(?=.*[$@$!%*#?&])[A-Za-z\d$@$!%*#?&]{8,}$", password):
                return JsonResponse({'message' : 'PASSWORD_VALIDATION_ERROR'}, status=400)
                
            if User.objects.filter(email=email).exists():
                return JsonResponse({'message' : 'DUPLICATED EMAIL'}, status=400)
```